test/nft-collection-list/nft-collection-list.test.ts

```diff
@@ -45,4 +50,40 @@ describe('NFTCollectionList', function () {
     expect(mainnetJson.networkId).toEqual(0)
     expect(testnetJson.networkId).toEqual(1)
   })
+
+  it('validate nft types', () => {
+    return Promise.all([
+      validateNftType(mainnetNFTCollectionList, mainnetURL),
+      validateNftType(testnetNFTCollectionList, testnetURL)
+    ])
+  })
+
+  it('validate nft metadata', () => {
+    return Promise.all([
+      validateNftMetadata(mainnetNFTCollectionList, mainnetURL),
+      validateNftMetadata(testnetNFTCollectionList, testnetURL)
+    ])
+  })
+
+  async function validateNftType(nftList: NFTCollectionList, url: string) {
+    const nodeProvider = new NodeProvider(url)
+
+    return Promise.all(
+      nftList.nftCollections.map((collection) =>
+        nodeProvider.guessFollowsNFTCollectionStd(collection.id).then((bool) => expect(bool).toEqual(true))
+      )
+    )
+  }
+
+  async function validateNftMetadata(nftList: NFTCollectionList, url: string) {
```

Comment:
Maybe it is worth following the URL and check if the returned json file complies with the following schema. WDTY?

https://github.com/alephium/alephium-web3/blob/master/packages/web3/std/nft_collection_interface.ral#L3-L23

Reply:
good idea

Reply:
since the [function return type](https://github.com/alephium/alephium-web3/blob/5c758a3ebd9f249f6b32082a04af195fcc9de7a0/packages/web3/src/api/node-provider.ts#L143) is already the [NFTCollectionMetadata](https://github.com/alephium/alephium-web3/blob/5c758a3ebd9f249f6b32082a04af195fcc9de7a0/packages/web3/src/api/types.ts#L285-L288), it's already checked by the type system no?

Reply:
I meant the schema pointed to by the `collectionUri`, which should contain `name`, `description` and `image`.

Reply:
aaaaaah that's a supra-great idea!!!

Reply:
I added a test, let me know if it was what you had in mind.